catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/ResourceOperations.java

```diff
@@ -411,6 +411,17 @@ ResourceResponse getResource(ResourceRequest resourceRequest, boolean isEnterpri
                 LOGGER.info("Unable to download resource", e);
             }
 
+            if (resourceResponse != null) {
+                // must add the request properties into response properties in case the source forgot to
```

Comment:
This method is already _huge_. Any chance we could move this to a private method?